models/management.cattle.io.user.js

```diff
@@ -17,6 +17,27 @@ export default {
     return !!(this.principalIds || []).find(p => p === currentPrincipal);
   },
 
+  async isAdmin() {
+    const canSeeSchemas = !!this.$store.getters[`management/schemaFor`](MANAGEMENT.GLOBAL_ROLE) && !!this.$store.getters[`management/schemaFor`](MANAGEMENT.GLOBAL_ROLE);
+
+    if (canSeeSchemas) {
+      // Cannot determine so can't be
+      return false;
+    }
+    const roles = await this.$store.dispatch('management/findAll', { type: MANAGEMENT.GLOBAL_ROLE });
+    const globalRoleBindings = await this.$store.dispatch('management/findAll', { type: MANAGEMENT.GLOBAL_ROLE_BINDING });
+
+    return !!globalRoleBindings
+      .find((binding) => {
+        if (binding.userName !== this.id) {
+          return false;
+        }
+        const globalRole = roles.find(r => r.id === binding.globalRoleName);
+
+        return globalRole.id === 'admin';
```

Comment:
This doesn't really work...  There's other built-in roles that are equivalent to admin as far as you need to be concerned for this (`restricted-admin`) and some users create all their own roles with names you won't know and don't use any of our built-in ones.  The only reliable method is to look for what permissions you actually need.

Reply:
With this one I was trying to determine between someone who would have all schema's compared to a user who's in the dynamic RBAC world (think this might include `restricted-admin`). @dramich In terms of users / principals / roles what's the best way to determine this?

Will rename the function to be more fitting or move the function entirely.

Reply:
Moved to #3720 (where it's been resolved)